### ISSUE TITLE
flowtype/require-types-at-top is thrown when mixing type/ declaration

### ISSUE THREAD

**Issue by gajus** [gajus/eslint-plugin-flowtype#373]
```js
declare function DataPoolConnectMethodType (): Promise<DatabasePoolConnectionType>;
declare function DataPoolConnectMethodType ((connection: DatabaseIsolatedPoolConnectionType) => Promise<void>): Promise<void>;

type FooType = string;

```

Expected:

No warnings.

Actual:

Getting "flowtype/require-types-at-top" error.

**Comment by gajus** [comment]
There is no way to overload a function without using `declare` (https://github.com/facebook/flow/issues/60).

We should therefore make declare and type compatible.
